jsx/src/components/GroupEdit/GroupEdit.jsx

```diff
@@ -67,38 +102,50 @@ const GroupEdit = (props) => {
             id="submit"
             className="btn btn-primary"
             onClick={() => {
-              // check for changes
-              if (!changed) {
-                history.push("/groups");
-                return;
-              }
-
               let new_users = selected.filter(
                 (e) => !group_data.users.includes(e)
               );
               let removed_users = group_data.users.filter(
                 (e) => !selected.includes(e)
               );
-
               let promiseQueue = [];
-              if (new_users.length > 0)
-                promiseQueue.push(addToGroup(new_users, group_data.name));
-              if (removed_users.length > 0)
-                promiseQueue.push(
-                  removeFromGroup(removed_users, group_data.name)
-                );
+              // check for changes
+              if (changed) {
+                if (new_users.length > 0)
+                  promiseQueue.push(addToGroup(new_users, group_data.name));
+                if (removed_users.length > 0)
+                  promiseQueue.push(
+                    removeFromGroup(removed_users, group_data.name)
+                  );
+              }
 
+              if (hasDuplicates(propkeys) == true) {
+                error.textContent = "Duplicate key found!";
+                error.style.color = "red";
+              } else {
+                //error.textContent = "";
+                propkeys.forEach((key, i) => (propobject[key] = propvalues[i]));
+              }
+              if (
+                propobject != group_data.properties &&
+                hasDuplicates(propkeys) == false
+              ) {
+                promiseQueue.push(updateProp(propobject, group_data.name));
```

Comment:
You could tie both the API call for group prop updates, and the API call for group user updates into one button click event that then returns the user to the `/groups` page. That might improve ease of use, but just an idea. @minrk thoughts?

Reply:
Both API calls for group prop and user updates are applied on the same button (Apply). I have only changed the way the button works, so that it commits both changes to user and prop and no longer sends the user back to the /groups page, in case the user would like to make an additional change before leaving the page. At the moment this button is the only feature that is conflicting with the main branch.